src/main/java/com/ncov/module/service/MaterialSuppliedService.java

```diff
@@ -68,6 +71,36 @@
         return carry(materials);
     }
 
+    @Transactional(rollbackFor = Exception.class)
+    public MaterialResponse update(Long materialId, MaterialRequest material, Long UserId) {
+        MaterialSuppliedEntity presentMaterial = getById(materialId);
+        if (!Objects.equals(presentMaterial.getMaterialSuppliedUserId(), UserId)) {
+            throw new AccessDeniedException("permission denied!");
+        }
+        MaterialDto materialDto = material.getMaterials().get(0);
+        AddressDto address = material.getAddress();
+        MaterialSuppliedEntity entity = MaterialSuppliedEntity.builder()
+                .id(materialId)
+                .country(address.getCountry())
+                .province(address.getProvince())
+                .city(address.getCity())
+                .district(address.getDistrict())
+                .streetAddress(address.getStreetAddress())
+                .materialSuppliedContactorName(material.getContactorName())
+                .materialSuppliedContactorPhone(material.getContactorPhone())
+                .materialSuppliedOrganizationName(material.getOrganisationName())
+                .materialSuppliedComment(material.getComment())
+                .materialSuppliedImageUrls(ImageUtils.joinImageUrls(materialDto.getImageUrls()))
+                .materialSuppliedName(materialDto.getName())
+                .materialSuppliedCategory(materialDto.getCategory())
+                .materialSuppliedQuantity(materialDto.getQuantity())
+                .materialSuppliedStandard(materialDto.getStandard())
+                .gmtModified(new Date())
```

Comment:
This will overwrite `gmtCreated` as null which should not be modified once persisted.